File: Array&String/String/CPP/string_matching_in_an_array.cpp
#include <algorithm>
#include <string>
#include <vector>
using namespace std;

bool comp(string a, string b) {
    return a.size() < b.size();
}

class Solution {
public:
    vector<string> stringMatching(vector<string>& words) {
        vector<string> res;
        sort(words.begin(), words.end(), comp);
        for (int i = 0; i < words.size(); i++) {
            for (int j = i+1; j < words.size(); j++) {
                if (words[j].find(words[i]) != -1) {
                    res.push_back(words[i]);
                    break;
                }
            }
        }
        return res;
    }
};

File: Array-String/String/Python/string_matching_in_an_array.py
class Solution:
    def stringMatching(self, words: list[str]) -> list[str]:
        words, res = sorted(words, key = lambda x: len(x)), []
        for i in range(len(words)):
            for j in range(i+1, len(words)):
                if words[i] in words[j]:
                    res.append(words[i])
                    break
        return res